tests/e2e/validatecluster/validatecluster_test.go

```diff
@@ -197,6 +198,41 @@ var _ = Describe("Verify Create", Ordered, func() {
 			}, "420s", "10s").Should(Succeed())
 		})
 
+		It("Verifies Restart K3s", func() {
```

Comment:
This claims to be just verifying that the service is restarted; what does the daemonset deployment have to do with verifying the restart?

Reply:
The daemonset is useful for making sure that non-system pods continue to work after a restart.